tests/network/vmi_istio.go

```diff
@@ -0,0 +1,313 @@
+		libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding(ports...)),
+		libvmi.WithLabel("app", vmiAppSelector),
+		libvmi.WithAnnotation(istioInjectSidecarAnnotation, "true"),
+	)
+	// Istio-proxy requires service account token to be mounted
+	tests.AddServiceAccountDisk(vmi, "default")
+	return vmi
+}
+
+var _ = SIGDescribe("[Serial] Istio", func() {
+	BeforeEach(func() {
+		if !istioServiceMeshDeployed() {
+			Skip("Istio service mesh is required for service-mesh tests to run")
+		}
+	})
+	var (
+		err        error
+		virtClient kubecli.KubevirtClient
+	)
+
+	Context("Virtual Machine with masquerade interface and explicitly specified ports", func() {
+		createJobCheckingVMIReachability := func(serverVMI *v1.VirtualMachineInstance, targetPort int) (*batchv1.Job, error) {
+			By("Starting HTTP Server")
+			tests.StartHTTPServer(vmi, targetPort)
+
+			By("Getting back the VMI IP")
+			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+			vmiIP := vmi.Status.Interfaces[0].IP
+
+			By("Running job to send a request to the server")
+			return virtClient.BatchV1().Jobs(tests.NamespaceTestDefault).Create(
+				context.Background(),
+				generateBatchJobSpec(vmiIP, fmt.Sprintf("%d", targetPort), tests.NewHelloWorldJobHTTP),
+				metav1.CreateOptions{},
+			)
+		}
+		BeforeEach(func() {
+			tests.BeforeTestCleanup()
+
+			virtClient, err = kubecli.GetKubevirtClient()
+			tests.PanicOnError(err)
+
+			// Enable sidecar injection by setting the namespace label
+			libnet.AddLabelToNamespace(tests.NamespaceTestDefault, tests.IstioInjectNamespaceLabel, "enabled")
+			defer libnet.RemoveLabelFromNamespace(tests.NamespaceTestDefault, tests.IstioInjectNamespaceLabel)
+
+			ports := []v1.Port{
+				{Port: svcDeclaredTestPort},
+				{Port: svcUndeclaredTestPort},
+			}
+			vmi = newVMIWithIstioSidecar(ports)
+			service = newService()
+
+			By("Create NetworkAttachmentDefinition")
+			nad = &k8snetworkplumbingwgv1.NetworkAttachmentDefinition{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "istio-cni",
+				},
+				Spec: k8snetworkplumbingwgv1.NetworkAttachmentDefinitionSpec{},
+			}
+			nad, err = virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(tests.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+
+			By("Creating VMI and associated Service")
+			_, err = virtClient.CoreV1().Services(tests.NamespaceTestDefault).Create(context.Background(), service, metav1.CreateOptions{})
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+
+			By("Waiting for VMI to be ready")
+			tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+		})
+		Describe("Inbound traffic", func() {
+			var job *batchv1.Job
```

Comment:
do you really need the `job` as a variable defined in this scope ? 

afaict, it could be locally scoped anytime you create it (since you do not have to clean it up manually in the test tear down.

Reply:
Done